File: Week 07/id_003/190.颠倒二进制位.js
/*
 * @lc app=leetcode.cn id=190 lang=javascript
 *
 * [190] 颠倒二进制位
 */

// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    let result = 0;
    for(let i=0; i < 32; i++){
        result = (result << 1) + (n & 1);
        n >>= 1;
    }
    return result >>> 0;
};
// @lc code=end

File: Week 07/id_158/190.颠倒二进制位.cs
/*
 * @lc app=leetcode.cn id=190 lang=csharp
 *
 * [190] 颠倒二进制位
 */

// @lc code=start
public class Solution
{
    public uint reverseBits(uint n)
    {
        uint res = 0;
        for (int i = 0; i < 32; i++)
        {
            res <<= 1;
            res += n & 1;
            n >>= 1;
        }
        return res;
    }
}
// @lc code=end

